### PR TITLE
feat: added 'APY' at the APR panel

### PR DESCRIPTION
**Pull request by impelcrypto**
**Pull Request Summary**

* feat: updated APR calculation formula
* feat: added 'APY' at the APR panel
* fix: modified the way the app changes the network
* feat: display 'Connect Wallet' button instead of 'Stake' button for users who haven't connect wallet yet
* refactor: removed `formatUnitAmount` function
  * ref:  https://github.com/AstarNetwork/astar.js/pull/22#discussion_r972164492
* fix: avoid not displaying anything when users go to other pages from assets page (especially on the localhost or without connect to wallets, this is a strange error) ([commit](https://github.com/AstarNetwork/astar-apps/pull/514/commits/57a1cb56e3cf27864b7e1c7588bb5874590f0de3))
  * gif (before): https://gyazo.com/d26d2afe1f3489f31f6d1afb3b2d45bd 

**Check list**
- [ ] contains breaking changes
- [ ] adds new feature
- [x] modifies existing feature (bug fix or improvements)
- [ ] relies on other tasks
- [ ] documentation changes
- [ ] tested on mobile devices

**This pull request makes the following changes:**

**Adds**

image

image

Gif: https://gyazo.com/4a9901cf167a3e5ec7af2a38b1d42e30

Ref
image

**Changes**

* feat: display 'Connect Wallet' button instead of 'Stake' button for users who haven't connect wallet yet
  * gif: https://gyazo.com/c954a3904ebaab8cc2777df6438ac6c8 

=Before=
image

=After=
image

### CODE REVIEW

**Review by Kahonnohak — APPROVED** [review]
Awesome work!

File: src/components/dapp-staking/DiscoverDappsTab.vue [hunk 1]
```diff
@@ -92,13 +99,15 @@ export default defineComponent({
...
 
     const maxNumberOfStakersPerContract = computed(
       () => store.getters['dapps/getMaxNumberOfStakersPerContract']
     );
-    const minimumStakingAmount = computed(() => {
+
+    const minimumStakingAmount = computed<string>(() => {
       const amount = store.getters['dapps/getMinimumStakingAmount'];
-      return formatUnitAmount(amount);
+      return Number(ethers.utils.formatEther(amount)).toString() + ' ' + nativeTokenSymbol.value;
```
Comment (hoonsubin) [inline src/components/dapp-staking/DiscoverDappsTab.vue:110]: Can we move these kinds of operations to astar.js? @niklabh

File: src/components/dapp-staking/statistics/Requirement.vue [hunk 2]
```diff
@@ -27,20 +27,22 @@
...
+    const { nativeTokenSymbol } = useNetworkInfo();
     const store = useStore();
     const maxNumberOfStakersPerContract = computed(
       () => store.getters['dapps/getMaxNumberOfStakersPerContract']
     );
-    const minimumStakingAmount = computed(() => {
+    const minimumStakingAmount = computed<string>(() => {
       const amount = store.getters['dapps/getMinimumStakingAmount'];
-      return formatUnitAmount(amount);
+      return Number(ethers.utils.formatEther(amount)).toString() + ' ' + nativeTokenSymbol.value;
```
Comment (hoonsubin) [inline src/components/dapp-staking/statistics/Requirement.vue:44]: same case as above

**Review by hoonsubin — APPROVED** [review]

File: src/config/chain.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  SHIDEN = 'Shiden',
+  ASTAR = 'Astar',
+  DEVELOPMENT = 'Development',
+}
+
+export type ASTAR_CHAIN = astarChain.SHIDEN | astarChain.ASTAR | astarChain.SHIBUYA;
+
+export type ASTAR_NETWORK_IDX = endpointKey.ASTAR | endpointKey.SHIDEN | endpointKey.SHIBUYA;
+
+export type ASTAR_EVM_NETWORK_IDX = 592 | 336 | 81;
```
Comment (bobo-k2) [inline src/config/chain.ts:14]: Can you check if we can get this form a node.

File: src/components/dapp-staking/DiscoverDappsTab.vue (hunk 1 above)
Comment (niklabh) [reply src/components/dapp-staking/DiscoverDappsTab.vue:110]: ok

File: src/config/chain.ts (hunk 3 above)
Comment (impelcrypto) [reply src/config/chain.ts:14]: This type is used for type annotation in [this variable](https://github.com/AstarNetwork/astar-apps/blob/main/src/hooks/useNetworkInfo.ts#L27) only. I don't think it is necessary to fetch it from node.

**Pull request merged**